packages/config/src/ProfileInfo.ts

```diff
@@ -39,6 +39,7 @@ import { CliUtils, ImperativeConfig } from "../../utilities";
 import { ImperativeError } from "../../error";
 import { ImperativeExpect } from "../../expect";
 import { Logger } from "../../logger";
+import { IProfArgAttrs } from "./doc/IProfArgAttrs";
 
 /**
  * This class provides functions to retrieve profile-related information.
```

Comment:
Is the new api all wrapped up in `ProfileInfo` class ? I don't want to be picky on names, but two initial reactions:
- `Profile` commands are being deprecated, should new APIs be associated with a Profile name?
- `Info` seems like it provides "profile info" but the description says it provides functions to retrieve profile-related information. Is this the final name?

Reply:
I was responsible for choosing the name ProfileInfo. Discussions with stakeholders made it clear that they wanted to limit retrieval of arguments to those arguments that were contained in profiles or defined within profile definitions. Even though questions about writing back continue to occur, our mandate was to read the information and merge any overriding values. Thus the name profile-information --> ProfileInfo. I just did not find a better name. I do not object to a better name. If we want to do that, I suggest that we approve and merge this PR, and open an issue for the name change to be completed before we deliver an @next version that can be used by VSCE validators.

Reply:
please see https://github.com/zowe/imperative/issues/582 :)